tests/subtractions/test_api.py

```diff
@@ -144,268 +112,344 @@ async def test_get_from_job(fake: DataFaker, spawn_job_client, snapshot_recent):
+        )
 
-    get_config_from_app(client.app).data_path = tmp_path
+        client = await spawn_job_client(authenticated=True)
 
-    user = await fake.users.create()
-    job = await fake.jobs.create(user)
+        data = {
+            "gc": {"a": 0.319, "t": 0.319, "g": 0.18, "c": 0.18, "n": 0.002},
+            "count": 100,
+        }
 
-    if exists:
-        await asyncio.gather(
-            mongo.subtraction.insert_one(
-                {
-                    "_id": "foo",
-                    "created_at": static_time.datetime,
-                    "file": {
-                        "id": 642,
-                        "name": "Apis_mellifera.1.fa.gz",
-                    },
-                    "name": "Foo",
-                    "nickname": "Foo Subtraction",
-                    "deleted": False,
-                    "ready": ready,
-                    "user": {"id": user.id},
-                    "job": {"job": job.id},
-                },
-            ),
-            mongo.samples.insert_one(
-                {"_id": "test", "name": "Test", "subtractions": ["foo"]},
-            ),
+        resp = await client.patch(f"/subtractions/{subtraction.id}", json=data)
+
+        assert resp.status == HTTPStatus.CONFLICT
+        assert await resp.json() == snapshot
+
+    async def test_finalize_subtraction_invalid_input_error(
+        self,
+        fake: DataFaker,
+        spawn_job_client: JobClientSpawner,
+        resp_is,
+    ):
+        user = await fake.users.create()
+        upload = await fake.uploads.create(
+            user=user,
+            upload_type=UploadType.subtraction,
+            name="foobar.fq.gz",
+        )
+        subtraction = await fake.subtractions.create(
+            user=user,
+            upload=upload,
+            finalized=True,
+        )
+
+        client = await spawn_job_client(authenticated=True)
+
+        resp = await client.patch(f"/subtractions/{subtraction.id}", data={})
+
+        await resp_is.invalid_input(
+            resp,
+            {"gc": ["required field"], "count": ["required field"]},
+        )
+
+
+class TestRemoveAsJob:
+    async def test_remove_ready_success(
+        self,
+        fake: DataFaker,
+        spawn_job_client: JobClientSpawner,
+        mongo: Mongo,
+        resp_is,
+    ):
+        user = await fake.users.create()
+        upload = await fake.uploads.create(
+            user=user,
+            upload_type=UploadType.subtraction,
+            name="foobar.fq.gz",
+        )
+        subtraction = await fake.subtractions.create(
+            user=user,
+            upload=upload,
+            finalized=True,
         )
 
-    resp = await client.delete("/subtractions/foo")
+        client = await spawn_job_client(authenticated=True)
+
+        await mongo.samples.insert_one(
+            {"_id": "test", "name": "Test", "subtractions": [subtraction.id]},
+        )
+
+        resp = await client.delete(f"/subtractions/{subtraction.id}")
```

Comment:
What happened to the assertions?

This request should return an error response since jobs are not allowed to delete finalized subtractions.